### PR TITLE
Add missing meta kwarg to gufunc class

### PR DESCRIPTION
**Pull request by pentschev**
- [ ] Attempts to address #7384
- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask`

### CODE REVIEW

**Comment by pentschev** [comment]
Reposting https://github.com/dask/dask/issues/7384#issuecomment-803060828 here to provide a brief explanation on what this PR is attempting to resolve:

> TBH, I'm not very skilled with gufuncs. However, I spent some (too much) time trying to understand what's going on here, I can confirm changes from #6736 are indeed responsible for this. The problem lies on those divergences between how libraries treat scalars, 0D and ND arrays, and I'm fairly certain this is to some extent very similar to the problem discussed in #6690 .
> 
> Right now, I don't have an optimal solution to offer, but a perhaps decently acceptable solution is to specify meta= in as_gufunc, as has been the trend for such corner cases that are really difficult to get working right among all different libraries. I opened #7423 to attempt providing a solution for this, but unfortunately I don't have the bandwidth now dig deeper on a solution that wouldn't require meta=.

**Comment by pentschev** [comment]
cc @TAdeJong @jsignell

**Review by jsignell — APPROVED** [review]
Thank you for writing this up @pentschev and sorry I didn't review it sooner. It seems to me that this is a clear improvement and if we want to finesse how other kwargs are handled in the future we can reasonably do that as well. Having the `meta` as a kwarg is a good escape hatch.

File: dask/array/gufunc.py [hunk 1]
```diff
@@ -560,6 +560,9 @@ class gufunc:
         dimensions are to consist only of one chunk.
         Warning: enabling this can increase memory usage significantly.
         Defaults to ``False``.
+    meta: Optional, tuple, keyword only
+        tuple of empty ndarrays describing the shape and dtype of the output of the gufunc.
```
Comment (TAdeJong) [inline dask/array/gufunc.py:564]:
```suggestion
        tuple of empty ndarrays describing the type of array and dtype of the output of the gufunc.
```

File: dask/array/gufunc.py [hunk 2]
```diff
@@ -602,6 +616,7 @@ def __init__(self, pyfunc, **kwargs):
         self.output_sizes = kwargs.pop("output_sizes", None)
         self.output_dtypes = kwargs.pop("output_dtypes", None)
         self.allow_rechunk = kwargs.pop("allow_rechunk", False)
+        self.meta = kwargs.pop("meta", None)
```
Comment (TAdeJong) [inline dask/array/gufunc.py:619]: Should we check, and throw a warning in case of, mismatch between `output_dtypes` and the dtypes as given in `meta`?

**Comment by TAdeJong** [comment]
@jsignell Oops, I had still noticed two small improvements to this PR... Just too late I guess.

**Comment by jsignell** [comment]
No worries, you are welcome to open another PR with these changes if you like :)

**Pull request merged**